waspc/examples/todoApp/main.wasp

```diff
@@ -5,7 +5,8 @@ app todoApp {
   title: "ToDo App",
   // head: [],
   webSocket: {
-    fn: import { webSocketFn } from "@src/webSocket",
+    fn: import { webSocketFn } from "@src/webSocket/index.js",
+    clientConfigFn: import { getConfig } from "@src/webSocket/clientConfigFn.js",
```

Comment:
You'll probably want to stick with this (since we already have a `webSocket` field), but just FYI.

Another possible approach is exposing a JS function that users could call in their client setup, like we do with `configureQueryClient`: https://wasp-lang.dev/docs/project/client-config#overriding-default-behaviour-for-queries.

Both options are both consistent and inconsistent with other parts of the API (it depends on what you compare them with).

Finally, does it make sense to take the middle ground?
For example, we expose a `transports` field in the Wasp file while also exposing a `configureWebSocket` hook users could use in the client setup function. This would make things easier for those who just want to change the `transport` option.

Reply:
I kinda dislike what I have now tbh, it feels weird to have it defined like that for some reason. 

The `configureQueryClient` approach might make more sense since it's a lower-level configuration than some other configuration e.g. `setupClient`. I'm not sure what to go for here.

Reply:
What about not doing it yet (see here: https://github.com/wasp-lang/wasp/pull/2205#pullrequestreview-2212096740)?

Reply:
Now I'm no longer sure. This PR came out of the feeling: _We are setting the transports to an opinionated default, we should allow customisation_ rather than coming as a request from our users. Maybe the best thing for now is to close the PR and see if users really want this. I guess by then we'll know more if we want to design it like `configureQueryClient` or as the client setup fn.

Reply:
> Maybe the best thing for now is to close the PR and see if users really want this.

I advise closing it, yes.

Another option is including this as a "shadow feature" (I made up the term, but it probably exists). We don't document the feature, we just put it in. Then, if someone requests it, we can "deploy" it right away. I'd say this makes sense for features that:
- Can have a high impact.
- We are certain someone will need.

But I don't think this one falls under those categories.

The developers (Wasp users) probably won't ever notice a problem. Their users might (if they're using ancient browsers), and will report it to the developers (who will probably tell them to update their browser, as they should anyway) :smile:

So yeah, I'd close it.

Reply:
Okay, let's keep it simple and close it. The impl is not that hard anyways, so when we see the need for this, we'll be able to quickly add it.